src/atomic_file/atomic.rs

```diff
@@ -116,14 +117,41 @@ impl AtomicFile {
         Ok(Self { directory, prefix })
     }
 
-    fn latest_version(&self) -> Result<usize> {
-        let mut max_version = 0;
-        for entry in fs::read_dir(&self.directory)? {
-            if let Some(version) = parse_version(&entry?.file_name()) {
-                max_version = std::cmp::max(max_version, version);
-            }
-        }
-        Ok(max_version)
+    /// Return a vec of files with latest version and the latest version. Multiples files can be found if they comes from different sources.
+    /// For example one from cellphone and one from computer can both a a version 2.
+    fn latest_version(&self) -> Result<(Vec<ReadOnlyFile>, usize)> {
+        let files_iterator = fs::read_dir(&self.directory)?.flatten();
+        let (files, version) = files_iterator.into_iter().fold(
+            (vec![], 0),
+            |(mut files, mut max_version), entry| {
+                let filename = entry.file_name();
+                if let Some(version) = parse_version(filename.to_str()) {
+                    // It's possible to have same version for two files coming from different machines
+                    // Add this files to the result
+                    if version >= max_version {
+                        let read_only = ReadOnlyFile {
+                            version,
+                            path: entry.path(),
+                        };
+                        files.push(read_only);
+                        max_version = version;
+                    }
```

Comment:
Does it really work correctly in case of conflicts for several versions?
E.g. versions: `ver1_a`, `ver2_a`, `ver3_a`, `ver3_b`, `ver4_a`, `ver4_b`, `ver5_a`.

It also can depend on the order we traverse version files, different order would give different results.

Reply:
@kirillt The order of parsing the files should not change anything. The list is filter after. I'll add some test

Reply:
The order should have no influence of the results since only files with latest version are returned